client_cli/src/main.rs

```diff
@@ -61,20 +62,27 @@ pub struct MetadataValueArg {
     /// The following types are supported:
     /// Numbers: decimal with optional point
     /// Booleans: false/true
-    /// JSON: e.g. {"Vec":[{"String":"a"},{"String":"b"}]}
+    /// Objects: e.g. {"Vec":[{"String":"a"},{"String":"b"}]}
     #[arg(short, long)]
-    value: MetadataValueBox,
+    value: JsonString,
 }
 
 impl FromStr for MetadataValueArg {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        s.parse::<bool>()
-            .map(MetadataValueBox::Bool)
-            .or_else(|_| s.parse::<Numeric>().map(MetadataValueBox::Numeric))
-            .or_else(|_| serde_json::from_str::<MetadataValueBox>(s).map_err(Into::into))
-            .map(|value| MetadataValueArg { value })
+        match Value::from_str(s)? {
```

Comment:
> create JsonString directly from String to have extra whitespaces

My main idea is to forbid creatin `JsonString` from a raw string to avoid invalid Jsons. Only `new(), from_str(), parse::<JsonString>()` are allowed because they implement ser/de chain "str->Value->String"